File: frontend/src/app/models/Piatto.ts
export class Piatto{
    id: number;
    nome: string;
    descrizione: string;
    prezzo: number;
    categoria: string;

    constructor(id: number, nome: string, descrizione: string, prezzo: number, categoria: string){
        this.id = id;
        this.nome = nome;
        this.descrizione = descrizione;
        this.prezzo = prezzo;
        this.categoria = categoria;
    }
}

File: backend/src/main/java/com/example/backend/models/Piatto.java
package com.example.backend.models;

import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import com.example.backend.models.Categoria;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "piatti")
@Data
public class Piatto{
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String nome;
    private String descrizione;
    private double prezzo;
    
    @Enumerated(EnumType.STRING)
    private Categoria categoria;
}